multiqc/utils/software_versions.py

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/env python
+
+""" Super Special-Case MultiQC module to produce report section on software versions """
+
+
+import logging
+import os
+from collections import defaultdict
+from textwrap import dedent
+
+import packaging.version
+import yaml
+
+from multiqc.modules.base_module import BaseMultiqcModule
+from multiqc.utils import report as mqc_report
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    def __init__(self):
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(
+            name="Software Versions",
+            anchor="multiqc_software_versions",
+            info="lists versions of software tools extracted from file contents.",
+        )
+
+        self.report_software_versions()
+
+    def report_software_versions(self):
+        """Create section listing software versions."""
+        content = self._make_versions_html(mqc_report.software_versions)
+        self.add_section(name=None, content=content)
+
+    @staticmethod
+    def _make_versions_html(versions):
+        """Generate a tabular HTML output of all versions."""
+        # source: https://github.com/nf-core/rnaseq/blob/3bec2331cac2b5ff88a1dc71a21fab6529b57a0f/modules/nf-core/custom/dumpsoftwareversions/templates/dumpsoftwareversions.py#L12
+        html = [
+            dedent(
+                """\
+                <table class="table mqc_versions_table">
+                    <thead>
+                        <tr>
+                            <th> Group </th>
```

Comment:
Need a MultiQC config option to be able to customise this, as the group could be different things to different users.